src/tests/ftest/pool/create_capacity.py

```diff
@@ -20,82 +19,123 @@ class PoolCreateCapacityTests(TestWithServers):
+                pool.create()
+                duration = time.time() - start
+
+                if duration > self.pool_create_duration:
+                    self.fail(
+                        f"Creating pool {index} took longer than "
+                        f"{self.pool_create_duration}s: got={duration}s")
+
+        except TestFail as error:
+            self.assertIn(
+                self.DER_NOSPACE, str(error),
+                f"Unexpected error occurred: wait={self.DER_NOSPACE}, got={str(error)}")
+
+            self.destroy_pools(self.pools[index])
+            self.pools = self.pools[:index]
+
+            self.assertGreaterEqual(
+                index, self.pool_quantity[0],
+                'Quantity of pools created should be greater or equal to '
+                f"{self.pool_quantity[0]}: got={index}")
+
+            self.log.info(
+                "Quantity of pools created lower than expected: wait=%i, got=%i",
+                self.pool_quantity[1], index)
+
     def test_create_pool_quantity(self):
         """JIRA ID: DAOS-5114 / SRS-2 / SRS-4.
 
         Test Description:
-            Create 200 pools on all of the servers.
+            Create a given number pools on all of the servers.
             Perform an orderly system shutdown via cmd line (dmg).
             Restart the system via cmd line tool (dmg).
             Verify that DAOS is ready to accept requests within 2 minutes.
+            Verify that all the created pools exists after the restart
 
         :avocado: tags=all,daily_regression
         :avocado: tags=hw,medium
         :avocado: tags=pool
         :avocado: tags=PoolCreateCapacityTests,test_create_pool_quantity
         """
-        # Create some number of pools each using a equal amount of 60% of the
-        # available capacity, e.g. 0.6% for 100 pools.
-        quantity = self.params.get("quantity", "/run/pool/*", 1)
-        storage = self.server_managers[0].get_available_storage()
-        if storage['nvme'] < 750156374016:
-            self.log.info(
-                'Reducing pool quantity from %s -> 150 due to insufficient NVMe capacity (%s < '
-                '750156374016)', quantity, storage['nvme'])
-            quantity = 150
-
         # Define all the pools with the same size defined in the test yaml
-        self.log_step('Defining {} pools'.format(quantity))
-        pools = []
-        for _ in range(quantity):
-            pools.append(add_pool(self, create=False))
-
-        # Create all the pools
-        self.log_step('Creating {} pools (dmg pool create)'.format(quantity))
-        self.get_dmg_command().server_set_logmasks("DEBUG", raise_exception=False)
-        check_pool_creation(self, pools, 30, 2)
-        self.get_dmg_command().server_set_logmasks(raise_exception=False)
-
-        # Verify DAOS can be restarted in less than 2 minutes
+        self.log_step('Creating {} pools'.format(self.pool_quantity[1]))
+        self.create_pools()
```

Comment:
To clarify the previous comment, I'm suggesting:
```suggestion
        # Define all the pools with the same size defined in the test yaml
        self.log_step('Defining {} pools'.format(quantity[1]))
        pools = []
        for _ in range(quantity[1]):
            pools.append(add_pool(self, create=False))

        # Create all the pools
        self.log_step('Attempt to creat {} pools (dmg pool create)'.format(quantity))
        self.get_dmg_command().server_set_logmasks("DEBUG", raise_exception=False)
        check_pool_creation(self, pools, 30, minimum=quantity[0])
        self.get_dmg_command().server_set_logmasks(raise_exception=False)
```

Then we could implement the minimum pool quantity check in `check_pool_creation`:
```
diff --git a/src/tests/ftest/util/test_utils_pool.py b/src/tests/ftest/util/test_utils_pool.py
index 421c988a0e..e124835dac 100644
--- a/src/tests/ftest/util/test_utils_pool.py
+++ b/src/tests/ftest/util/test_utils_pool.py
@@ -112,7 +112,7 @@ def get_size_params(pool):
             "nvme_size": pool.nvme_per_rank}


-def check_pool_creation(test, pools, max_duration, offset=1, durations=None):
+def check_pool_creation(test, pools, max_duration, offset=1, durations=None, minimum=None):
     """Check the duration of each pool creation meets the requirement.

     Args:
@@ -122,11 +122,21 @@ def check_pool_creation(test, pools, max_duration, offset=1, durations=None):
         offset (int, optional): pool index offset. Defaults to 1.
         durations (list, optional): list of other pool create durations to include in the check.
             Defaults to None.
+        minimum (int, optional): minimum number of pools that must be created if specified.
     """
     if durations is None:
         durations = []
     for index, pool in enumerate(pools):
-        durations.append(time_pool_create(test.log, index + offset, pool))
+        try:
+            durations.append(time_pool_create(test.log, index + offset, pool))
+        except TestFail as error:
+            if minimum is None:
+                raise error
+            if "DER_NOSPACE(-1007)" not in str(error):
+                raise error
+            if index < minimum:
+                test.fail(
+                    "Minimum pool quantity ({}/{}) not reached: {}".format(index, minimum, error))

     exceeding_duration = 0
     for index, duration in enumerate(durations):

```

Reply:
Fixed with commit 72578e892fa07df1b2e6c4b0a1fea17c10f6ed7d